runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -4012,11 +4096,11 @@ MM_CopyForwardScheme::clearMarkMapForPartialCollect(MM_EnvironmentVLHGC *env)
 					if (_extensions->tarokEnableExpensiveAssertions) {
 						Assert_MM_true(_markMap->checkBitsForRegion(env, region));
 					}
-				} else if (region->hasValidMarkMap()) {
-					void *low = region->getLowAddress();
-					void *bumpPointer = ((MM_MemoryPoolBumpPointer *)region->getMemoryPool())->getAllocationPointer();
-					void *high = (void *)MM_Math::roundToCeiling(CARD_SIZE, (UDATA)bumpPointer);
-					_markMap->setBitsInRange(env, low, high, true);
+//				} else if (region->hasValidMarkMap()) {
```

Comment:
Hm I wonder if this is correct... by clearing the whole markmap of this region, we will clear some valid markmap bits for existing objects (not that I can tell right away why we would need those bits).

Reply:
This looks safe afterall. I misunderstood original code is clearing alloc to top of survivor regions but it was clearing markmap from base to alloc of evac regions. 
It's suboptimal thought, so please do add a todo comment: consider remembering where the last allocated object is, to minimize clearing for regions with low occupancy (indeed, regions with low occupancy are rather good candidates for evacuation).